backend/lib/services/SpaceListing/interface.ts

```diff
@@ -0,0 +1,14 @@
+import { SpaceListing } from '@prisma/client';
+
+
+export interface ISpaceListingService {
+    getAllSpaceListings(): Promise<SpaceListing[]>
+    getSpaceListingById({ spaceListingId }: { spaceListingId: string }):
```

Comment:
nit: we probably don't need to call it spaceListingId (can just call it id) if the getSpaceListingById function is so expressive

```ts
    getSpaceListingById(id: number):
```